prov/rxm/src/rxm_cq.c

```diff
@@ -1384,6 +1384,14 @@ void rxm_ep_do_progress(struct util_ep *util_ep)
 				buf, repost_entry);
 
 		/* Discard rx buffer if its msg_ep was closed */
+		if (!buf->conn) {
+			assert(buf->ep->srx_ctx);
+			buf->conn = rxm_key2conn(buf->ep,
+						 buf->pkt.ctrl_hdr.conn_id);
+			if (OFI_UNLIKELY(!buf->conn))
+				continue;
```

Comment:
Thanks for the fix! We would probably need to add `assert(buf->conn);` as well since a NULL buf->conn would be a bug in this case.

Reply:
> Thanks for the fix! We would probably need to add `assert(buf->conn);` as well since a NULL buf->conn would be a bug in this case.

If I put the assert after rxm_key2conn(), it appears that buf->conn will sometimes show up as NULL. Any ideas?

Reply:
Sorry, I misread your patch at first. When shared rx context is used, we should just re-post recv using rx_buf->msg_ep which would be set to the shared rx context fid. There's no need to free the buffer here until the shared rx context is closed. So, I think the below lines need to be modified as:
```
if (!rxm_ep->srx_ctx && !buf->conn->msg_ep) {
	ofi_buf_free(&buf->hdr);
	continue;
}
```

Reply:
> Sorry, I misread your patch at first. When shared rx context is used, we should just re-post recv using rx_buf->msg_ep which would be set to the shared rx context fid. There's no need to free the buffer here until the shared rx context is closed. So, I think the below lines need to be modified as:
> 
> ```
> if (!rxm_ep->srx_ctx && !buf->conn->msg_ep) {
> 	ofi_buf_free(&buf->hdr);
> 	continue;
> }
> ```

Thanks, I'll test and update the PR!

Reply:
@a-ilango - Updated the PR. That makes sense; I should have looked a little closer at the intent of the original PR. Thanks.